compiler/rustc_resolve/src/build_reduced_graph.rs

```diff
@@ -18,6 +18,7 @@ use rustc_expand::base::ResolverExpand;
 use rustc_expand::expand::AstFragment;
 use rustc_hir::def::{self, *};
 use rustc_hir::def_id::{CRATE_DEF_ID, DefId, LocalDefId};
+use rustc_index::bit_set::DenseBitSet;
```

Comment:
We often have PRs that change DenseBitSet (formerly BitSet) to MixedBitSet
- https://github.com/rust-lang/rust/pull/134438
- https://github.com/rust-lang/rust/pull/142471

So it might be worth trying that out also, depending on perf results.

Reply:
I think that's unlikely to be a good fit, since there's one bit per macro rule, and since the bits get cleared. You would need a massive number of rules. I expect the common case to fit in a single machine word with room to spare.

Doesn't mean someone shouldn't try it, but I would be surprised if it made a difference unless DenseBitSet is somehow very inefficient.

Reply:
Yeah, I only really mean it as an alternative if *somehow* the perf results come back weird-as-in-bad.